### PR TITLE
Add Golang SQLite demo

### PR DESCRIPTION
**Pull request by james-dong-security**
Add Golang SQLite demo

### CODE REVIEW

File: demos/golang/go_sqlite/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Use Golang and SQLite with Occlum
+
+This project demonstrates how Occlum enables [Golang](https://golang.org) programs with [SQLite](https://www.sqlite.org/index.html) library calls running in SGX enclaves. The way to use a SQL (or SQL-like) database in Go is through the [database/sql](https://golang.org/pkg/database/sql/) package. The SQL package exposes an ideal set of generic APIs for a variety of SQL or SQL-like databases. The SQL package itself must be used in conjunction with a [database driver](https://github.com/golang/go/wiki/SQLDrivers) package.
+
+The demo program is based on a widely used [sqlite3 driver](https://github.com/mattn/go-sqlite3) conforming to the built-in database/sql interface and passing the compatibility test suite at [https://github.com/bradfitz/go-sql-test](https://github.com/bradfitz/go-sql-test). You can run the web server demo on Occlum via
```
Comment (tatetian) [inline demos/golang/go_sqlite/README.md:5]: `the web server demo` should be changed to `the SQLite demo`.

File: demos/golang/go_sqlite/run_go_sqlite_demo.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	.process.default_stack_size = "40MB" |
+	.process.default_heap_size = "320MB" |
+	.process.default_mmap_size = "960MB" ' Occlum.json)" && \
+echo "${new_json}" > Occlum.json
+
+# Copy program into Occlum Workspace and build
+cp ../simple_demo image/bin
+occlum build
+
+# Run the hello world sample
```
Comment (tatetian) [inline demos/golang/go_sqlite/run_go_sqlite_demo.sh:36]: `Run the hello world sample` should be corrected.

File: demos/golang/go_sqlite/run_go_sqlite_demo.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+NC='\033[0m'
+
+# Install SQLite with occlum-go
+occlum-go get -u -v github.com/mattn/go-sqlite3
+
+# Build the Golang SQLite demo program using the Occlum Golang toolchain(i.e., occlum-go)
+occlum-go build -o simple_demo simple_demo.go
+
+sqlite_demo="./simple_demo"
+if [ ! -f $sqlite_demo ];then
```
Comment (tatetian) [inline demos/golang/go_sqlite/run_go_sqlite_demo.sh:14]: I don't think this check is necessary. If the `occlum-go build` command failed, then the script should have exited (due to `set -e`).

**Review by tatetian — COMMENTED** [review]
Thanks for the PR! And, sorry for my late comments. The commit looks good to me in general. I have found some minor issues that I hope you can address.

File: demos/golang/go_sqlite/README.md (hunk 1 above)
Comment (james-dong-security) [reply demos/golang/go_sqlite/README.md]: Done. Thanks for catching it!

File: demos/golang/go_sqlite/run_go_sqlite_demo.sh (hunk 2 above)
Comment (james-dong-security) [reply demos/golang/go_sqlite/run_go_sqlite_demo.sh]: Done. Thanks.

File: demos/golang/go_sqlite/run_go_sqlite_demo.sh (hunk 3 above)
Comment (james-dong-security) [reply demos/golang/go_sqlite/run_go_sqlite_demo.sh]: Done. That is great suggestion. Thanks.

**Comment by james-dong-security** [comment]
@tatetian 
Thanks for reviewing this patch. Made changes based on your comments. New patch has been uploaded.
PTAL

**Review by tatetian — APPROVED** [review]

**Comment by jessehui** [comment]
@james-dong-security Hi, James. Thank you for contributing this demo to our project. Could you also enable CI for this demo as well? A separate patch is fine. Thank you in advance.

**Comment by james-dong-security** [comment]
@jessehui :
Would love to enable the CI test as well. Will send out a patch soon. Thanks for proposing it.

**Pull request merged**